### PR TITLE
Add doc for performance tests

### PR DESCRIPTION
**Pull request by MaxThevenet**
This needs some more testing, at at some point maybe someone else could try it.
@atmyers I remember you were looking into it at some point. Did you develop a similar feature for other AMReX codes?

### CODE REVIEW

File: Docs/source/developers/performance_tests.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+will submit this job once, and all the following ones. It will:
+
+ - Create directory ``$SCRATCH/performance_warpx`` if doesn't exist.
+ - Create 1 sub-directory per week per number of nodes (1,2,8,64,256,512).
+ - Submit one job per number of nodes. It will run 6 different tests, each twice (to detect fluctuations).
+ - Submit an analysis job, that will read the results ONLY AFTER all runs are finished. This uses the dependency feature of the batch system.
+ - This job reads the Tiny Profiler output for each run, and stores the results in a pandas file at the hdf5 format.
+ - Execute ``write_csv.py`` from the ``perf_logs`` repo to append a csv and a hdf5 file with the new results.
+ - Commit the results (but DOES NOT PUSH)
```
Comment (ax3l) [inline Docs/source/developers/performance_tests.rst]:
```suggestion
 - Commit the results (but DO NOT PUSH YET)
```

File: Docs/source/developers/performance_tests.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Setup on Summit @ OLCF
+----------------------
+
+Here's my setup on Summit, feel free to adapt it to your needs:
+
+.. code-block:: sh
+
+   mkdir $HOME/AUTOMATED_PERF_TESTS
+   export AUTOMATED_PERF_TESTS=$HOME/AUTOMATED_PERF_TESTS/ # I put it in $HOME/my_bashrc.sh
+   export SCRATCH=/gpfs/alpine/scratch/<user name>/aph114/ # I put it in $HOME/my_bashrc.sh
```
Comment (ax3l) [inline Docs/source/developers/performance_tests.rst]: Just a tiny idea for generalization:
```suggestion
   # I put the next three lines in $HOME/my_bashrc.sh
   export proj=aph114  # project for job submission
   export AUTOMATED_PERF_TESTS=$HOME/AUTOMATED_PERF_TESTS/
   export SCRATCH=/gpfs/alpine/scratch/$(whoami)/$proj/
   
   mkdir $HOME/AUTOMATED_PERF_TESTS
```

File: Docs/source/developers/performance_tests.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+   source $HOME/my_bashrc.sh
+
+   # Make sure all dependencies are install (in $HOME/.local/)
+   cd $HOME
+   module load python/3.7.0-anaconda3-5.3.0
+   pip install numpy==1.15.4 --user
+   pip install pandas --user
+   pip install bokeh --user
+   pip install gitpython --user
```
Comment (ax3l) [inline Docs/source/developers/performance_tests.rst]: If we place this in an environment we can avoid that this interacts with other workflows:
```suggestion
   # Make sure all dependencies are installed and loaded
   cd $HOME
   module load python/3.7.0
   python3 -m pip install --user --upgrade pip
   python3 -m pip install --user virtualenv
   python3 -m venv $HOME/sw/venvs/warpx-perftest
   source $HOME/sw/venvs/warpx-perftest/bin/activate
   python3 -m pip install numpy==1.15.4
   python3 -m pip install pandas
   python3 -m pip install bokeh
   python3 -m pip install gitpython
```

File: Docs/source/developers/performance_tests.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+   # Check where numpy is installed, and prepend its parent directory to PYTHONPATH
+   home_to_numpy=$(python -c "import numpy ; import os ; print(os.path.split(os.path.split(numpy.__file__)[0])[0])")
```
Comment (ax3l) [inline Docs/source/developers/performance_tests.rst]: Interesting, is this still needed? If you can `import` numpy it means it is already in a PYTHONPATH or python system path.
```suggestion
   home_to_numpy=$(python3 -c "import numpy ; import os ; print(os.path.split(os.path.split(numpy.__file__)[0])[0])")
```

**Review by ax3l — COMMENTED** [review]
This is really cool, I don't know anyone that has such a great history of performance over time for a rationally complex project. Super cool, thank you for creating, curating and sharing this!

File: Docs/source/developers/performance_tests.rst (hunk 4 above)
Comment (MaxThevenet) [reply Docs/source/developers/performance_tests.rst]: This is still required for me, but I am not sure why.

File: Docs/source/developers/performance_tests.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+WarpX has automated performance test scripts, which run weak scalings for various tests on a weekly basis. The results are stored in the `perf_logs repo <https://github.com/ECP-WarpX/perf_logs>`_ and plots of the performance history can be found on `this page <https://ecp-warpx.github.io/perf_logs/>`_.
+
+These performance tests run automatically, so they need to do ``git`` operations etc. For this reason, they need a separate clone of the source repos, so they don't conflict with one's usual operations. This is typically in a sub-directory in the ``$HOME``, with variable ``$AUTOMATED_PERF_TESTS`` pointing to it. Similarly, a directory is needed to run the simulations and store the results. By default, it is ``$SCRATCH/performance_warpx``.
+
+The test runs a weak scaling (1,2,8,64,256,512 nodes) for 6 different tests ``Tools/PerformanceTests/automated_test_{1,2,3,4,5,6}_*``, gathered in 1 batch job per number of nodes to avoid submitting too many jobs.
+
+Setup on Summit @ OLCF
+----------------------
+
+Here's my setup on Summit, feel free to adapt it to your needs:
```
Comment (LDAmorim) [inline Docs/source/developers/performance_tests.rst]:
```suggestion
Here is an example setup for Summit:
```

File: Docs/source/developers/performance_tests.rst [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+.. code-block:: sh
+
+   # I put the next three lines in $HOME/my_bashrc.sh
+   export proj=aph114  # project for job submission
```
Comment (LDAmorim) [inline Docs/source/developers/performance_tests.rst:20]:
```suggestion
   export proj=<PorjectID>  # project for job submission
```
I thought we always tried to use the projectID (for privacy), like you did in the libEnsemble example in Tools.
Is it not important?

File: Docs/source/developers/performance_tests.rst [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+   git clone https://github.com/ECP-WarpX/picsar.git
+   git clone --branch development https://github.com/AMReX-Codes/amrex.git
+   git clone https://github.com/ECP-WarpX/perf_logs.git
+
+Then, in ``$AUTOMATED_PERF_TESTS``, create a file ``run_automated_performance_tests_512.sh`` with the following content:
+
+.. code-block:: sh
+
+   #!/bin/bash -l
+   #BSUB -P APH114
```
Comment (LDAmorim) [inline Docs/source/developers/performance_tests.rst:36]:
```suggestion
   #BSUB -P <ProjectID>
```

File: Docs/source/developers/performance_tests.rst [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+   #BSUB -W 00:10
+   #BSUB -nnodes 1
+   #BSUB -J PERFTEST
+   #BSUB -e err_automated_tests.txt
+   #BSUB -o out_automated_tests.txt
+
+   source $HOME/my_bashrc.sh
+
+   # Make sure all dependencies are installed and loaded
+   cd $HOME
```
Comment (LDAmorim) [inline Docs/source/developers/performance_tests.rst:46]: Can we add a comment with something like "Alternatively create a `conda` environment and load it in the job script"?

Comment (LDAmorim) [reply Docs/source/developers/performance_tests.rst:46]: Or `pip` of course!

**Review by LDAmorim — COMMENTED** [review]
Thanks for this docs @MaxThevenet !
They are very helpful and encourage testing, which is great!

File: Docs/source/developers/performance_tests.rst (hunk 8 above)
Comment (MaxThevenet) [reply Docs/source/developers/performance_tests.rst:46]: TBH I am not totally confident this setting works out of the box, so it is a bit premature to suggest an alternative.

File: Docs/source/developers/performance_tests.rst (hunk 6 above)
Comment (MaxThevenet) [reply Docs/source/developers/performance_tests.rst:20]: This page is meant to be used by WarpX developers, so I would be OK to hard-code the project name for now.

**Review by LDAmorim — APPROVED** [review]

**Comment by MaxThevenet** [comment]
I removed the WIP tag, as I think this can already be merged.

**Review by ax3l — APPROVED** [review]
Love this very much, thank you! :sparkles:

**Pull request merged**